### PR TITLE
no same did you mean

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Fixes:  #6780

It's a quick and a little dirty fix, because dataframe's command is never executed, but I think the error message is better than misleading message.  Current result:

```
❯ [1 2] | as-datetime
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #16:1:1]
 1 │ [1 2] | as-datetime
   ·         ─────┬─────
   ·              ╰── executable was not found
   ╰────
  help: No such file or directory (os error 2)
```

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace --features=extra` to check that all the tests pass

# Documentation

- [x] If your PR touches a user-facing nushell feature then make sure that there is an entry in the documentation (https://github.com/nushell/nushell.github.io) for the feature, and update it if necessary.

### CODE REVIEW

File: crates/nu-protocol/src/shell_error.rs [hunk 1]
```diff
@@ -816,8 +816,17 @@ pub fn did_you_mean(possibilities: &[String], input: &str) -> Option<String> {
     let suggestion =
         crate::lev_distance::find_best_match_for_name_with_substrings(&possibilities, input, None)
             .map(|s| s.to_string());
+
+    // Note: sometimes plugin or external command define's the same command
+    // as user input, but the command doesn't match nu's requirements
+    //
+    // like [1 2] | to-datetime, `find_best_match_for_name_with_substrings`
+    // will return `to-datetime`, but that's absolutely not what user want.
```
Comment (sholderbach) [inline crates/nu-protocol/src/shell_error.rs:824]:
```suggestion
    // will return `to-datetime`, but is not compatible as it is a dataframe command.
```

File: crates/nu-protocol/src/shell_error.rs [hunk 2]
```diff
@@ -852,8 +861,8 @@ mod tests {
             (
                 vec!["OS", "PWD", "PWDPWDPWDPWD"],
                 vec![
-                    ("pwd", Some("PWD"), "Exact case insensitive match yields a match"),
-                    ("pwdpwdpwdpwd", Some("PWDPWDPWDPWD"), "Exact case insensitive match yields a match"),
+                    ("pwd", None, "The same item doesn't means match"),
+                    ("pwdpwdpwdpwd", None, "The same item doesn't means match"),
```
Comment (sholderbach) [inline crates/nu-protocol/src/shell_error.rs:865]: I am not sure what you mean here.

File: crates/nu-protocol/src/shell_error.rs [hunk 3]
```diff
@@ -863,7 +872,7 @@ mod tests {
                 vec!["foo", "bar", "baz"],
                 vec![
                     ("fox", Some("foo"), ""),
-                    ("FOO", Some("foo"), ""),
+                    ("FOO", None, ""),
```
Comment (sholderbach) [inline crates/nu-protocol/src/shell_error.rs:875]: Will this lead to correct behavior if the user has mistyped the case of a command (or maybe an env var)?

Comment (WindSoilder) [reply crates/nu-protocol/src/shell_error.rs:875]: Thanks, that's a case that I never thought about..

File: crates/nu-protocol/src/shell_error.rs (hunk 2 above)
Comment (WindSoilder) [reply crates/nu-protocol/src/shell_error.rs]: Sorry for my poor english, but I've change the case, it should yield a match

**Comment by WindSoilder** [comment]
Sorry, the solution for this is not greated, caused we can type `AS-datetime`, and nusuggested `as-datetime`, then we type `as-datetime`, nu said something not found.   Closing this for now
